mcp3008/mcp3008.go

```diff
@@ -0,0 +1,92 @@
+// Package mcp3008 implements a driver for the MCP3008 Analog to Digital Converter.
+//
+// Datasheet: http://ww1.microchip.com/downloads/en/DeviceDoc/21295d.pdf
+//
+package mcp3008 // import "tinygo.org/x/drivers/mcp3008"
+
+import (
+	"errors"
+	"machine"
+)
+
+// Device wraps MCP3008 SPI ADC.
+type Device struct {
+	bus machine.SPI
+	cs  machine.Pin
+	tx  []byte
+	rx  []byte
+	CH0 *ADCPin
+	CH1 *ADCPin
+	CH2 *ADCPin
+	CH3 *ADCPin
+	CH4 *ADCPin
+	CH5 *ADCPin
+	CH6 *ADCPin
+	CH7 *ADCPin
+}
+
+// ADCPin is the implementation of the ADConverter interface.
+type ADCPin struct {
+	machine.ADC
```

Comment:
Why do you need to embed this type? Why not just `machine.Pin`?

Reply:
A future PR will add an interface type `ADConverter` so we can use the same as a normal `machine.ADC` for other drivers such as the `thermistor`.

Reply:
But you wouldn't need to embed this type to implement an interface?

For example, the interface could look like this:

```go
type ADCer interface {
    Get() uint16
}
```

In that case, the `ADCPin` type already implements the interface.
(Also, perhaps `Get() uint16` is a bit too generic if ADC is used as an interface).

Reply:
Changed to simply `machine.Pin`